uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -195,6 +214,11 @@ define([
         }
 
         vm.error = undefined;
+        //update leaverequest
+        if (updateRequest()) {
```

Comment:
This needs to be refactored, it's not good currently for several reason

The name of the function, combined with the fact that it is invoked in an `if` statement makes this code hard to understand. What does it mean _"if updateRequest then return"_? Is the function simply returning a boolean? Is it checking for something? Is it also doing an update? Why is it returning a boolean at all, given the name? You have to go and read the function implementation to find out

Why are we returning at all, in the middle of `vm.submit`, basically having a major branching between updating and creating a leave request? Can't this be done in a more elegant way? A leave request update and a leave request create are basically the same thing:
1. you send the current leave request status to the server
2. when the server is done, you emit an event
3. you cancel any `error`
4. you call `vm.ok()`
5. If any error happens during the process, you `catch` it and store the error message in `vm.error`

Given the above, is there any need for the branching at all? This surely can be improved